platform/ios/src/MGLMapboxEvents.m

```diff
@@ -543,7 +543,17 @@ - (void) postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             [request setHTTPBody:jsonData];
 
             // Send non blocking HTTP Request to server
-            [[_session dataTaskWithRequest:request] resume];
+            if (strongSelf.session) {
+                [[strongSelf.session dataTaskWithRequest:request] resume];
+            } else {
+                // Session is invalidated when telemetry is paused, which means
+                // we have to manually manage its lifecycle here for the turnstile.
+                strongSelf.session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]
+                                                                   delegate:strongSelf
+                                                              delegateQueue:nil];
+                [[strongSelf.session dataTaskWithRequest:request] resume];
+                [strongSelf.session finishTasksAndInvalidate];
```

Comment:
Rather than checking for the existence of `MGLMapboxEvents.session`, I think it’d be better to check for paused state, or better yet, for an indication that this is a turnstile event. If we’re getting in here without a session for anything other than a turnstile event, that’s an error on our part, so the original code attempts to fail gracefully.

Additionally, could we keep this synchronous, turnstile-specific session a local variable, to avoid confusion about what kind of session is being stored in the ivar?